tests/trial_tests/test_fixed_frozen.py

```diff
@@ -0,0 +1,253 @@
+            trial.suggest_categorical("x", [{"foo": "bar"}])  # type: ignore
+
+
+@parametrize_trial_type
+def test_not_contained_param(trial_type: type) -> None:
+    trial = _create_trial(
+        trial_type=trial_type,
+        params={"x": 1.0},
+        distributions={"x": FloatDistribution(1.0, 10.0)},
+    )
+    with pytest.warns(UserWarning):
+        assert trial.suggest_float("x", 10.0, 100.0) == 1.0
+
+    trial = _create_trial(
+        trial_type=trial_type,
+        params={"x": 1.0},
+        distributions={"x": FloatDistribution(1.0, 10.0, log=True)},
+    )
+    with pytest.warns(UserWarning):
+        assert trial.suggest_float("x", 10.0, 100.0, log=True) == 1.0
+
+    trial = _create_trial(
+        trial_type=trial_type,
+        params={"x": 1.0},
+        distributions={"x": FloatDistribution(1.0, 10.0, step=1.0)},
+    )
+    with pytest.warns(UserWarning):
+        assert trial.suggest_float("x", 10.0, 100.0, step=1.0) == 1.0
+
+    trial = _create_trial(
+        trial_type=trial_type,
+        params={"x": 1.0},
+        distributions={"x": IntDistribution(1, 10)},
+    )
+    with pytest.warns(UserWarning):
+        assert trial.suggest_int("x", 10, 100) == 1
+
+    trial = _create_trial(
+        trial_type=trial_type,
+        params={"x": 1},
+        distributions={"x": IntDistribution(1, 10)},
+    )
+    with pytest.warns(UserWarning):
+        assert trial.suggest_int("x", 10, 100, 1) == 1
+
+    trial = _create_trial(
+        trial_type=trial_type,
+        params={"x": 1},
+        distributions={"x": IntDistribution(1, 10, log=True)},
+    )
+    with pytest.warns(UserWarning):
+        assert trial.suggest_int("x", 10, 100, log=True) == 1
+
+
+@parametrize_trial_type
+def test_set_user_attrs(trial_type: type) -> None:
+
+    trial = _create_trial(trial_type)
+    trial.set_user_attr("data", "MNIST")
+    assert trial.user_attrs["data"] == "MNIST"
+
+
+@parametrize_trial_type
+def test_set_system_attrs(trial_type: type) -> None:
+
+    trial = _create_trial(trial_type)
+    trial.set_system_attr("system_message", "test")
+    assert trial.system_attrs["system_message"] == "test"
+
+
+@parametrize_trial_type
+def test_report(trial_type: type) -> None:
+
+    # FrozenTrial ignores reported values.
+    trial = _create_trial(trial_type)
+    trial.report(1.0, 1)
+    trial.report(2.0, 2)
+
+
+@parametrize_trial_type
+def test_should_prune(trial_type: type) -> None:
+
+    # FrozenTrial never prunes trials.
+    assert _create_trial(trial_type).should_prune() is False
+
+
+@parametrize_trial_type
+def test_datetime_start(trial_type: type) -> None:
+
+    trial = _create_trial(trial_type)
+    assert trial.datetime_start is not None
+    old_date_time_start = trial.datetime_start
+    trial.datetime_complete = datetime.datetime.now()  # type: ignore
+    assert trial.datetime_complete != old_date_time_start  # type: ignore
```

Comment:
I think we can remove `# type: ignore` by asserting them not `None`.

Reply:
Hmm.. I see. Can we fix the type annotation as `Union[FixedTrial, FrozenTrial]` to remove the `# type: ignore`?

Reply:
No, it didn't work. I think mypy is not smart enough to infer that.

Reply:
It seems that the `FixedTrial` does not have `datetime_complete`. How about just asserting as follows?
```suggestion
    assert datetime.datetime.now() != old_date_time_start
```

Reply:
Thanks, I'll apply your suggestion.